share/spice/bitcoin/bitcoin.js

```diff
@@ -1,90 +1,104 @@
-            prices = api_result[currency];
-        } else if (params.length > 1) {
-            // Just allow the currency as a parameter
+        if (!api_result || api_result.status !== 200) {
             return Spice.failed('bitcoin');
-        } else {
-            currency = $.trim(params[0].toUpperCase());
-            prices = api_result[currency];
-
-            if (!prices) {
-                // Is not a valid currency
-                return Spice.failed('bitcoin');
-            }
         }
+        
+        
+        var script = $('[src*="/js/spice/bitcoin/"]')[0],
+            source = $(script).attr("src"),
+            query = decodeURIComponent(source.match(/bitcoin\/([a-z]{0,12})/)[1]),
+            api_result = api_result.data;
 
-        var buy = {
-            formatted_price: getFormattedPrice(currency, prices.buy, prices.symbol),
-            title: "BUY"
+        var formatBtc = function(amount) {
+            return (amount / 100000000.0) + " BTC"
         };
+        
+        DDG.require('moment.js', function(){
+            var spice = {
+                meta: {
+                    sourceName: "Biteasy"
+                },
+                templates: {
+                    options:{
+                        content: 'record',
+                        moreAt: true,
+                        rowHighlight: true
+                    }
+                },
+            };
 
-        var sell = {
-            formatted_price: getFormattedPrice(currency, prices.sell, prices.symbol),
-            title: "SELL"
-        };
+            switch (query) {
+                case "addresses":
+                    spice.id = "bitcoin_address";
+                    spice.name = "Bitcoin Address";
+                    spice.templates.group = "info";
+                    
+                    spice.data = {
+                        "record_data": {
+                            "Current Balance": formatBtc(api_result.balance),
+                            "Total Received": formatBtc(api_result.total_received),
+                            "Total Sent": formatBtc(api_result.total_sent),
+                            "Hash160": api_result.hash160
+                        }
+                    };
+                    spice.normalize = function(item) {
+                        return {
+                            image: 'https://www.biteasy.com/blockchain/addresses/' + api_result.address + '/qrcode.jpeg',
+                            img_m: 'https://www.biteasy.com/blockchain/addresses/' + api_result.address + '/qrcode.jpeg'
+                        };
+                    };
+                    spice.meta.sourceUrl = "https://www.biteasy.com/blockchain/addresses/" + api_result.address;
+                    break;
+                    
+                case "transactions":
+                    spice.id = "bitcoin_transaction";
+                    spice.name = "Bitcoin Transaction";
+                    spice.templates.group = "base";
```

Comment:
If we switch this to the `list` group, we can get rid of almost all the CSS :)

The BitcoinAddress Spice does show a QR code though which may not work with the list group. It could potentially be the `{{content}}` and then you'd need a bit of CSS to float it to the left. I'm not sure how that will look on mobile though, but it may help to simplify this IA.